packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -0,0 +1,938 @@
+  public abstract readonly securityGroupId: string;
+
+  public abstract export(): DatabaseInstanceImportProps;
+
+  /**
+   * Defines a CloudWatch event rule which triggers for instance events. Use
+   * `rule.addEventPattern(pattern)` to specify a filter.
+   */
+  public onEvent(id: string, target?: events.IEventRuleTarget, options?: events.EventRuleProps) {
+    const rule = new events.EventRule(this, id, options);
+    rule.addEventPattern({
+      source: [ 'aws.rds' ],
+      resources: [
+        this.node.stack.formatArn({
+          service: 'rds',
+          resource: 'db',
+          sep: ':',
+          resourceName: this.instanceIdentifier
+        })
+      ]
+    });
+    rule.addTarget(target);
+    return rule;
+  }
+
+  /**
+   * Renders the secret attachment target specifications.
+   */
+  public asSecretAttachmentTarget(): secretsmanager.SecretAttachmentTargetProps {
+    return {
+      targetId: this.instanceIdentifier,
+      targetType: secretsmanager.AttachmentTargetType.Instance
+    };
+  }
+}
+
+/**
+ * The engine for the database instance.
+ */
+export enum DatabaseInstanceEngine {
+  Aurora = 'aurora',
+  AuroraMysql = 'aurora-mysql',
+  AuroraPostgresql = 'aurora-postgresql',
+  MariaDb = 'mariadb',
+  Mysql = 'mysql',
+  OracleEE = 'oracle-ee',
+  OracleSE2 = 'oracle-se2',
+  OracleSE1 = 'oracle-se1',
+  OracleSE = 'oracle-se',
+  Postgres = 'postgres',
+  SqlServerEE = 'sqlserver-ee',
+  SqlServerSE = 'sqlserver-se',
+  SqlServerEX = 'sqlserver-ex',
+  SqlServerWeb = 'sqlserver-web'
+}
+
+/**
+ * The license model.
+ */
+export enum LicenseModel {
+  /**
+   * License included.
+   */
+  LicenseIncluded = 'license-included',
+
+  /**
+   * Bring your own licencse.
+   */
+  BringYourOwnLicense = 'bring-your-own-license',
+
+  /**
+   * General public license.
+   */
+  GeneralPublicLicense = 'general-public-license'
+}
+
+/**
+ * The processor features.
+ */
+export interface ProcessorFeatures {
+  /**
+   * The number of CPU core.
+   */
+  readonly coreCount?: number;
+
+  /**
+   * The number of threads per core.
+   */
+  readonly threadsPerCore?: number;
+}
+
+/**
+ * The type of storage.
+ */
+export enum StorageType {
+  /**
+   * Standard.
+   */
+  Standard = 'standard',
+
+  /**
+   * General purpose (SSD).
+   */
+  GP2 = 'gp2',
+
+  /**
+   * Provisioned IOPS (SSD).
+   */
+  IO1 = 'io1'
+}
+
+/**
+ * The retention period for Performance Insight.
+ */
+export enum PerformanceInsightRetentionPeriod {
+  /**
+   * Default retention period of 7 days.
+   */
+  Default = 7,
+
+  /**
+   * Long term retention period of 2 years.
+   */
+  LongTerm = 731
+}
+
+/**
+ * Construction properties for a DatabaseInstanceNew
+ */
+export interface DatabaseInstanceNewProps {
+  /**
+   * The name of the compute and memory capacity classes.
+   */
+  readonly instanceClass: ec2.InstanceType;
+
+  /**
+   * Specifies if the database instance is a multiple Availability Zone deployment.
+   *
+   * @default false
```

Comment:
What's the implications of making this `true` by default? Wouldn't that be a better practice?

Reply:
Yes, if possible I would like to a have production defaults in this constructs: db.r5.large, multi az, 1000 IOPS, 100 GiB, encryption enabled with default master key, 7 days backup, enhanced monitoring, no auto minor version upgrade (the console experience when selecting _Production_).

Looking at the discussion we had here https://github.com/awslabs/aws-cdk/pull/2063#discussion_r267537358 and the issue you raised with _implicit costs to customer_, would you be ok to accept these production defaults?

Reply:
@eladb any update on whether production defaults are acceptable for this construct?

Reply:
That's an interesting conversation, and I wonder if this would be something we run into in the future, sort of "presets". The RDS console definitely demonstrates that there is value in allowing people to being configuring their new database from a predefined set of options that are tailored for a use case.

Maybe we should have a similar experience:

```ts
new rds.DatabaseInstance(this, 'instance', rds.DatabaseInstancePresets.production());
// or, override some value:
new rds.DatabaseInstance(this, 'instance', rds.DatabaseInstancePresets.production({
  multiAz: false
}));
```

Something like that?

Reply:
How would you specify mandatory properties like `engine` and `masterUsername` with this API? Spread the preset and add those properties? 
```ts
{
  ...rds.DatabaseInstancePresets.production(),
  masterUsername: '...'
}
```